virttest/migration.py

```diff
@@ -271,26 +283,196 @@ def do_migration(self, vms, srcuri, desturi, migration_type,
         :param thread_timeout: time out seconds for the migration thread running
         :param ignore_status: determine if an exception is raised for errors
         :param func: the function executed during migration thread is running
+        :param multi_funcs: list of functions executed during migration
+                     thread is running. The func and multi_funcs should not be
+                     provided at same time. For example,
+                     multi_funcs = [{"func": <function check_established at 0x7f5833687510>,
+                                     "after_event": "iteration: '1'",
+                                     "before_event": "Suspended Migrated",
+                                     "before_pause": "yes", "func_param": params},
+                                    {"func": <function domjobabort at 0x7f5835cd08c8>,
+                                     "before_pause": "no"}
+                                   ]
         :param args: dictionary used by func,
-                     'func_param' is mandatory if no real func_param, none is
-                     requested.
+                     'func_param' is mandatory for func parameter.
+                     If no real func_param, none is requested.
                      'shell' is optional, where shell=True(bool) can be used
                      for process.run
-
         """
+
+        def _run_collect_event_cmd():
```

Comment:
@dzhengfy 
It looks like quite number of embedded methods are newly introduced.
Do you think whether it  is necessary to move them into separate python file ,and let migration.py to use by import?

Reply:
@chunfuwen , these embedded functions are closely related to do_migration() function, so I think there is useless for other functions to reuse them. I prefer to keep them here.

Reply:
Currently seems the embedded functions not that common to move to other file, so I think it is ok to just keep them here.